components/net/cookie_storage.rs

```diff
@@ -229,14 +229,19 @@ fn reg_host<'a>(url: &'a str) -> String {
 
 fn is_cookie_expired(cookie: &Cookie) -> bool {
     match cookie.expiry_time {
-        Some(ref t) => t.to_timespec() <= time::get_time(),
+        Some(ref t) => {
+            t.duration_since(UNIX_EPOCH).unwrap_or_default() <=
+                SystemTime::now()
+                    .duration_since(UNIX_EPOCH)
+                    .unwrap_or_default()
+        },
```

Comment:
Can't these just be compared directly with`t < SystemTime.now()`?

Reply:
less than or equal is not implement for SystemTime

Reply:
It seems like it is though? https://doc.rust-lang.org/std/time/struct.SystemTime.html#impl-PartialOrd-for-SystemTime

Reply:
Ah sorry serde misleaded me, should be `t.0 <= SystemTime::now()`